File: Problems/No.1829_Maximum_XOR_for_Each_Query/maximum_xor_for_each_query.py
class Solution(object):
    def getMaximumXor(self, nums, maximumBit):
        """
        :type nums: List[int]
        :type maximumBit: int
        :rtype: List[int]
        """

        # parameters
        num_Size = len(nums)
        previous_xor = 0
        MAX_k = (1<<maximumBit) - 1
        MAX_XOR = [0] * num_Size

        # start computing
        for i in range(num_Size):
            previous_xor ^= nums[i]
            MAX_XOR[num_Size-i-1] = previous_xor ^ MAX_k
        
        return MAX_XOR

File: Problems/No.1829_Maximum_XOR_for_Each_Query/maximum_xor_for_each_query.cpp
class Solution {
public:
    vector<int> getMaximumXor(vector<int>& nums, int maximumBit) {
        
        // parameters
        int update_location = nums.size() - 1;
        int previous_xor = 0;
        const int MAX_k = (1<<maximumBit) - 1;
        std::vector<int> MAX_XOR(nums.size(), 0);

        // start computing
        for (auto &num: nums) {

            previous_xor ^= num;
            MAX_XOR[update_location--] = previous_xor ^ MAX_k;

        }

        return MAX_XOR;

    }
};
